mlir/lib/Dialect/SPIRV/IR/SPIRVOps.cpp

```diff
@@ -972,8 +972,73 @@ void spirv::FuncOp::print(OpAsmPrinter &printer) {
 }
 
 LogicalResult spirv::FuncOp::verifyType() {
-  if (getFunctionType().getNumResults() > 1)
+  FunctionType fnType = getFunctionType();
+  if (fnType.getNumResults() > 1)
     return emitOpError("cannot have more than one result");
+
+  auto funcOp = dyn_cast<spirv::FuncOp>(getOperation());
+
+  auto hasDecorationAttr = [&](spirv::Decoration decoration,
+                               unsigned argIndex) {
+    for (auto argAttr :
+         cast<FunctionOpInterface>(*funcOp).getArgAttrs(argIndex)) {
+      if (auto decAttr = dyn_cast<spirv::DecorationAttr>(argAttr.getValue()))
```

Comment:
Ok, I add checking the attribute name.

> It would be nice to add an IR test for such case to ignore #spirv.decoration attributes without spirv.decoration name.

I think that an attribute of an argument requires the decoration name, so this IR test expects an error `error: expected attribute name` for this case ( sets `#spirv.decoration` attributes without `spirv.decoration` name).

Reply:
ah I meant adding a test like `%arg0: !spirv.ptr<..> { random_attr = #spirv.decoration<Aliased> }`.